org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/AdvancedNewProjectPage.java

```diff
@@ -0,0 +1,489 @@
+		return enu.toString().equals(combo.getText());
+	}
+
+	protected <T extends Control> void reportIssue(int severity, String text) {
+		reportIssue(severity, text, () -> {});
+	}
+
+	protected <T extends Control> void reportIssue(int severity, String text, Procedure0 fix) {
+		statusWidget.setStatus(severity, text, fix, () -> {
+			validate(null);
+		});
+		((Composite) getControl()).layout();
+	}
+
+	protected <T extends Control> void addIssue(int severity, String text) {
+		statusWidget.addStatus(severity, text);
+		((Composite) getControl()).layout();
+	}
+
+	protected boolean isBundleResolved(String bundleId) {
+		Bundle[] bundles = Activator.getInstance().getBundle().getBundleContext().getBundles();
+		Optional<Bundle> bundle = Arrays.stream(bundles).filter(b -> bundleId.equals(b.getSymbolicName())).findFirst();
+		return bundle.isPresent() && (bundle.get().getState() & (Bundle.RESOLVED | Bundle.STARTING) | Bundle.ACTIVE) != 0;
+	}
+
+	protected Group Group(Composite parent, Procedure1<? super Group> config) {
+		Group group = new Group(parent, SWT.NONE);
+		group.setFont(parent.getFont());
+		group.setLayout(new GridLayout(1, false));
+		group.setLayoutData(new GridData(SWT.FILL, SWT.TOP, true, false));
+		config.apply(group);
+		return group;
+	}
+
+	protected Button CheckBox(Composite composite, Procedure1<? super Button> config) {
+		Button button = new Button(composite, SWT.CHECK);
+		button.setFont(button.getParent().getFont());
+		button.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
+		config.apply(button);
+		return button;
+	}
+
+	protected Button Radio(Composite composite, Procedure1<? super Button> config) {
+		Button button = new Button(composite, SWT.RADIO);
+		button.setFont(button.getParent().getFont());
+		button.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
+		config.apply(button);
+		return button;
+	}
+
+	protected Combo DropDown(Composite parent, Procedure1<? super Combo> config) {
+		Combo combo = new Combo(parent, SWT.READ_ONLY);
+		combo.setFont(parent.getFont());
+		combo.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+		config.apply(combo);
+		return combo;
+	}
+
+	protected ControlDecoration InfoDecoration(Control control, String text) {
+		FieldDecoration infoField = FieldDecorationRegistry.getDefault().getFieldDecoration(FieldDecorationRegistry.DEC_INFORMATION);
+		ControlDecoration controlDecoration = new ControlDecoration(control, (SWT.TOP + SWT.RIGHT));
+		controlDecoration.setImage(infoField.getImage());
+		controlDecoration.setDescriptionText(text);
+		controlDecoration.setShowHover(true);
+		return controlDecoration;
+	}
+
+	protected void setDefaults() {
+		createUiProject.setSelection(true);
+		createIdeProject.setSelection(true);
+		createTestProject.setSelection(true);
+		createWebProject.setSelection(false);
+		createSDKProject.setSelection(false);
+		createP2Project.setSelection(false);
+
+		select(preferredBuildSystem, BuildSystem.NONE);
```

Comment:
The original code gets the enum using `BuildSystem.values.head`. I have replaced it with `BuildSystem.NONE`. Similar change is done in the below two lines.

Reply:
LGTM